c/termination-numeric-2/lychrel_false-termination.c

```diff
@@ -0,0 +1,35 @@
+extern int __VERIFIER_nondet_int(void);
+
+long number, reversedNr = 0;
+
+long reverse() {
+  long remainder;
+  reversedNr = 0;
+
+  while (number > 0) {
+    remainder = number % 10;
+    reversedNr = (reversedNr * 10) + remainder;
+    number /= 10;
```

Comment:
This will lead to `number` being equal to `0` after the call to `reverse()`. Then `number = reversedNr + number;` will simply replace number by the reverse of its (former) value. This is surely not what is intended here, i.e. the value of number should rather be stored in a local variable that can then be safely modified. Same will hold for the other task (`lychrel_true-termination.c`)

Reply:
I do not see the problem here. The loop already aborts when number is zero. Can you give an example?

Reply:
The problem is not that the loop does not abort. It does abort, as you say. But the semantics of the function reverse() for the Lychrel problem should be to return the reverse of `number` without the side effect of setting the global variable `number` to zero. Line 24 should assign `number` to the sum of its previous value and the reverse of its previous value.

Example: Assume that `number` has the value 1234. Then line 24 should assign the value 1234+4321 to `number`. But because of the side effect of `reverse()`, `number` is set to 0 first, so the new value of `number` will be 0+4321.
In other words: The label `false-termination` is correct here, but the task does not represent the actual lychrel problem. Doing so is not even possible using `int` as datatype because of potential overflows.